src/forklift/plugin.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# * coding: utf8 *
+'''
+plugin.py
+
+A module the contains the plugin class that should be inherited from when building plugins
+'''
+
+import logging
+import settings
+from os.path import join
+
+
+class ScheduledUpdateBase(object):
+
+    def __init__(self):
+        #: the logging module to keep track of the plugin
+        self.log = logging.getLogger(settings.LOGGER)
+        #: the duration in hours that marks a plugin as dirty
+        self.expires_in_hours = 24
+        #: the path to where the data in dependencies resides
+        self.source_location = 'C:\\MapData\\SGID10.sde'
+        #: the parent path to where the output will be created
+        self.output_directory = 'C:\\MapData\\'
+        #: the file geodatabase where data will be inserted
+        self.gdb_name = 'SGID10.gdb'
+        #: the table names for all dependent data for an application
+        self.dependencies = []
+
+    def nightly(self):
+        '''This method will be called by forklift if the `expires_in_hours` value has expired.
+        This is the only method that needs to be implemented and will otherwise throw a `NotImplementedError`.
+        '''
+
+        raise NotImplementedError('Implement nightly in your child plugin.')
```

Comment:
Then maybe we shouldn't force the plugin to implement it? Then it could be something like, if you have any additional logic that needs to be run other than the one-to-one update then implement the nightly method.